src/H5Ipublic.h

```diff
@@ -80,7 +80,7 @@ typedef int64_t hid_t;
  * can be removed from the ID type. If the function returns negative
  * (failure) then the object will remain in the ID type.
  */
-typedef herr_t (*H5I_free_t)(void *);
+typedef herr_t (*H5I_free_t)(void *, void **);
```

Comment:
This is going to have to be handled like H5L_iterate_t and compatibility with 1.12.0 is going to be an issue.